### PR TITLE
Dashboard with Edge Device health (presence) metrics

### PR DESCRIPTION
**Pull request by jakub-dzon**
Follow-up for #146 adding Grafana dashboard panels for monitoring edge devices presence.

### CODE REVIEW

**Comment by masayag** [comment]
I wasn't able to open the dashboard in the browser with 40k devices.
Also, grafana's pod complains about:
```
t=2022-03-27T11:46:06+0000 lvl=eror msg="Request error" logger=context userId=1 orgId=1 uname=admin error="net/http: abort Handler"
t=2022-03-27T11:46:11+0000 lvl=eror msg="Request error" logger=context userId=1 orgId=1 uname=admin error="net/http: abort Handler"
t=2022-03-27T11:46:16+0000 lvl=eror msg="Request error" logger=context userId=1 orgId=1 uname=admin error="net/http: abort Handler"
```
The health panel will not report anything, while others do. 
Selection_1087

**Comment by bardielle** [comment]
When I ran the deployment script it failed with the error- csv created in namespace with multiple operatorgroups, can't pick one automatically. I used that cmd- ```tools/deploy_grafana.sh docs/metrics/flotta-dashboard.json```

**Comment by jakub-dzon** [comment]
> I wasn't able to open the dashboard in the browser with 23k devices. Also, grafana's pod complains about:

Warned you:)That's what I was expecting for large numbers of devices - panels for all devices are good for smaller numbers of them, but with larger numbers it's better to limit scope by filtering device IDs or namespaces. 
That said - it worked OK-ish for me with that number of devices when the time range was small (like 10,15, 30 minutes).

So - maybe a separate dashboard JSON with just these two panels plus a doc would be better?

**Comment by jakub-dzon** [comment]
> When I ran the deployment script it failed with the error- csv created in namespace with multiple operatorgroups, can't pick one automatically. I used that cmd- `tools/deploy_grafana.sh docs/metrics/flotta-dashboard.json`

I think that's unrelated to the dashboard changes. @masayag - WDYT?

**Comment by masayag** [comment]
> So - maybe a separate dashboard JSON with just these two panels plus a doc would be better?

It will be more useful for high scale. A parametrized dashboards that allows to select namespace will be more efficient in that case, e.g. how it is done in https://grafana.com/grafana/dashboards/11671
In our case namespace & time range can be configurable.

**Comment by masayag** [comment]
> > When I ran the deployment script it failed with the error- csv created in namespace with multiple operatorgroups, can't pick one automatically. I used that cmd- `tools/deploy_grafana.sh docs/metrics/flotta-dashboard.json`
> 
> I think that's unrelated to the dashboard changes. @masayag - WDYT?

Agree. that's an issue with the script that was developed and tested against SNO only in which there were no operatorgroup under flotta namespace.

**Comment by jakub-dzon** [comment]
@masayag - I've created a dedicated dashboard with filter box allowing users to pick desired deviceNamespace.

**Review by masayag — APPROVED** [review]

File: docs/metrics/flotta-devices-health.json [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+      "points": false,
+      "renderer": "flot",
+      "seriesOverrides": [],
+      "spaceLength": 10,
+      "stack": false,
+      "steppedLine": false,
+      "targets": [
+        {
+          "exemplar": true,
+          "expr": "ceil(sum by(deviceID, namespace)(rate(flotta_operator_edge_devices_heartbeat{job=~\"flotta.+\"}[2m])))",
```
Comment (masayag) [inline docs/metrics/flotta-devices-health.json:237]: should this be changed to `deviceNamespace` as well?

**Review by masayag — COMMENTED** [review]
Removing ack since noticed the right panel doesn't preview data

File: docs/metrics/flotta-devices-health.json (hunk 1 above)
Comment (masayag) [reply docs/metrics/flotta-devices-health.json:237]: This one with changing the query for the right pane:
Selection_1088

Comment (jakub-dzon) [reply docs/metrics/flotta-devices-health.json:237]: Yes!

Comment (jakub-dzon) [reply docs/metrics/flotta-devices-health.json]: Done

**Review by masayag — APPROVED** [review]

**Review by bardielle — APPROVED** [review]

**Pull request merged**